discovery-provider/src/queries/get_app_name_metrics.py

```diff
@@ -1,12 +1,124 @@
 import logging
 import time
-from sqlalchemy import func, desc
-
-from src.models import AppNameMetrics
+import functools as ft
+from datetime import date, timedelta
+from sqlalchemy import func, desc, asc
+from src import exceptions
+from src.models import AppNameMetrics, DailyAppNameMetrics, MonthlyAppNameMetrics
 from src.utils import db_session
 
 logger = logging.getLogger(__name__)
 
+def get_historical_app_metrics():
+    """
```

Comment:
let's keep an eye on the response size here, but it should probably be totally fine for a long time